nextcloud/Caddyfile

```diff
@@ -28,6 +28,11 @@ my-nextcloud-site.com {
 		to /remote.php/{1}/{2}
 	}
 
+	rewrite {
+		r ^/public.php/(.+?)(\/?)$
+		to /public.php/(.+?)(\/?)$
```

Comment:
It perform requests to public.php when someone shared something by link. Confirming it on 11.0.3

See screenshot.
<img width="620" alt="2017-07-16 21 12 14" src="https://user-images.githubusercontent.com/869402/28249248-7e0d4aa2-6a6b-11e7-8e0f-c672261e7f9f.png">